### ISSUE TITLE
Optional chaining parsing failed

### ISSUE THREAD

**Issue by shabinpshams** [SonarSource/SonarJS#1990]
Typescript = 3.7.5

INFO: SonarQube Scanner 3.2.0.1227
INFO: Java 1.8.0_121 Oracle Corporation (64-bit)
INFO: Linux 4.1.12-94.3.9.el7uek.x86_64 amd64
INFO: SonarQube server 8.2.0

ERROR: Failed to parse the file.

**Comment by vilchik-elena** [comment]
Please be so kind to provide us with the logs of the analysis and the reproducer file

**Comment by shabinpshams** [comment]
> Please be so kind to provide us with the logs of the analysis and the reproducer file

I am getting an error in Jenkins while parsing files

ERROR: Failed to parse file [file path] at line 76: Expression expected.

And in the file on the same location, I could see optional chaining (a?.b?.c). And is supported only in typescript 3.7.5.

**Comment by vilchik-elena** [comment]
For parsing we use TypeScript installed in your system. 

Are you sure you have version >=3.7.5 installed locally during analysis?

I tried locally, on my side when I'm using 3.7.5 version no parsing error is logged.

**Comment by ld210** [comment]
We're facing the same issue, i'm getting the error "Expected an assignment or function call and instead saw an expression" where optional chaining is used. 
The installed TS version is 3.7.5

**Comment by P3tronius** [comment]
Same issue here, `ERROR: Failed to parse file [file path] at line XX: Expression expected.`
We have typescript 3.8.3 in our `package.json`and are using task `Sonar Maven 3`in Bamboo.

**Comment by fuzzykiller** [comment]
We have the same issue on VSTS. This is purely a parse error. It does not result in any error/warning/code smell in Sonar Cloud.

"Run Code Analysis" task v1.13.0
SonarScanner 4.3.0.2102
Java 11.0.7 Azul Systems, Inc. (64-bit)
Linux 5.4.0-1012-azure amd64

(Not sure how to get the SonarJS version.)

Relevant log output:
```
INFO: Sensor TypeScript analysis [javascript]
INFO: Using TypeScript at: '/agent/_work/2/s/source/Schoeck.Applikation.Web/ClientApp/node_modules/@angular-devkit/build-optimizer/node_modules'
INFO: Found 1 tsconfig.json file(s): [/agent/_work/2/s/source/Schoeck.Applikation.Web/ClientApp/tsconfig.json]
INFO: 45 source files to be analyzed
INFO: Analyzing 45 files using tsconfig: /agent/_work/2/s/source/xxx/ClientApp/tsconfig.json
##[error]ERROR: Failed to parse file [xxx/ClientApp/src/app/components/canvas/canvas.component.ts] at line 19: Expression expected.
```

Line 19 contains:
```ts
this.points$ = this.sourcePoints$?.pipe(map((points) => this.transformPoints(points)));
```

This is an Angular project that compiles just fine.

`package.json` contains:

```js
{
  // ...
  "devDependencies": {
    // ...
    "typescript": "^3.7.5"
  }
}
```

**Comment by vilchik-elena** [comment]
@fuzzykiller could you confirm that location '/agent/_work/2/s/source/Schoeck.Applikation.Web/ClientApp/node_modules/@angular-devkit/build-optimizer/node_modules' contains typescipt >=3.7.5?

**Comment by fuzzykiller** [comment]
Sorry, I made a mistake while removing sensitive information from my log. Could you please also remove from your comment? Thanks!

---

This folder contains TypeScript 3.6.5, so an older version.

**Comment by vilchik-elena** [comment]
@fuzzykiller indeed. ticket is there already https://github.com/SonarSource/SonarJS/issues/1928
gonna fix it soon

**Comment by yassin-kammoun-sonarsource** [comment]
Optional chaining parsing is supported in recent SonarQube versions.

**Comment by JuanCSoto** [comment]
Hello @yassin-kammoun-sonarsource I´m in Version 8.9.8 (build 54436) and i have the error

![image](https://user-images.githubusercontent.com/17452860/170368615-dbdb6ab0-f23a-4fec-8f7f-5162813cb014.png)

**Comment by yassin-kammoun-sonarsource** [comment]
Hey @JuanCSoto, I answered you in another thread. As mentioned, please report the problem you are facing on https://community.sonarsource.com.

Thank you :)
